### PR TITLE
Add separate packages for eslint and prettier configs

### LINKED ISSUE #3: Publish separate packages for eslint and prettier configs
### Description

Right now, this repository published one package `kwizapp/eslint-config` which holds both configuration files for prettier and eslint. 

### Tasks
To make this more manageable, the optimal setup would be the following:
- [x] rename this repository into `kwiz-dotfiles`
- [x] within `kwiz-dotfiles` there are the following two folders
  - [x] `eslint-configs/**`
  - [x] `prettier-config/`
- [x] each of these subfolders have their own `package.json` and publish different packages
- [x] `eslint/` publishes `@kwizapp/eslint-config/**`
- [x] `prettier/` published `@kwizapp/prettier-config`
- [x] add READMEs
- [x] `dependencies` inside `package.json` based on ranges
- [x] update workflow to not publish new package with each push

### PR DESCRIPTION
**Pull request by nikzaugg**
### Description

This PR add the following:
* npm 3 packages for `eslint` configs for `Javascript`, `Typescript` and `Typescript React`
* npm package for `prettier` config

### Checklist

- [x] Have you added tests where necessary? Do all the test pass? 
- [x] Have you added descriptive comments to your code?
- [x] Have you updated the documentation related to this propo

### CODE REVIEW

**Comment by nikzaugg** [comment]
Hey guys

I think it's almost ready. 

I still don't really know how to best handle the whole releasing cycle etc. 
Because now, all everything is failing, because it cannot release over an existing version of the package, even if you add a new commit. The important part is the `version` inside `package.json`.

Let's try to find a good solution here.

File: eslint-configs/javascript/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+### Linting
+
+Make sure you enable linting inside your editor.
+
+**`npm scripts`**
+You can use the following command to lint your project files. Add the following to your `package.json`.
+
+```json
+scripts": {
+  "lint": "eslint \"{src,test}/**/*.{js,ts,tsx}\" --fix",
```
Comment (rschlaefli) [inline eslint-configs/javascript/README.md:59]: We can't lint ts and tsx with this package only (or can, but won't) ;)

File: eslint-configs/typescript-react/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+### Linting
+
+Make sure you enable linting inside your editor.
+
+**`npm scripts`**
+You can use the following command to lint your project files. Add the following to your `package.json`.
+
+```json
+scripts": {
+  "lint": "eslint \"{src,test}/**/*.{ts,tsx}\" --fix",
```
Comment (rschlaefli) [inline eslint-configs/typescript-react/README.md:62]: Why should we not lint js here?

File: eslint-configs/typescript/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+### Linting
+
+Make sure you enable linting inside your editor.
+
+**`npm scripts`**
+You can use the following command to lint your project files. Add the following to your `package.json`.
+
+```json
+scripts": {
+  "lint": "eslint \"{src,test}/**/*.{ts,tsx}\" --fix",
```
Comment (rschlaefli) [inline eslint-configs/typescript/README.md:61]: Lint js as well (there might be some files that need to be left as .js)

Comment (rschlaefli) [reply eslint-configs/typescript/README.md:61]: That would then just apply the base config as imported from the other package

**Review by rschlaefli — APPROVED** [review]
Great work, thanks a lot! I think it will be very helpful.

I suggest using lerna for the releasing stuff, if it is not too complex and time-consuming to setup. We should create an issue and discuss further.

File: eslint-configs/javascript/README.md (hunk 1 above)
Comment (nikzaugg) [reply eslint-configs/javascript/README.md:59]: updated here https://github.com/kwizapp/kwiz-dotfiles/packages/160122

File: eslint-configs/typescript-react/README.md (hunk 2 above)
Comment (nikzaugg) [reply eslint-configs/typescript-react/README.md:62]: updated description here https://github.com/kwizapp/kwiz-dotfiles/packages/160123

File: eslint-configs/typescript/README.md (hunk 3 above)
Comment (nikzaugg) [reply eslint-configs/typescript/README.md:61]: updated description here: https://github.com/kwizapp/kwiz-dotfiles/packages

**Comment by nikzaugg** [comment]
> Great work, thanks a lot! I think it will be very helpful.
> 
> I suggest using lerna for the releasing stuff, if it is not too complex and time-consuming to setup. We should create an issue and discuss further.

Yes, I opened a new issue here https://github.com/kwizapp/kwiz-dotfiles/issues/5 for the future.

**Pull request merged**